### ISSUE TITLE
User autocomplete: The slug has too low color contrast

### ISSUE THREAD

**Issue by carolinan** [WordPress/gutenberg#69001]
### Description

When you type `@` in the editor, a list of users is shown.
The list shows the avatar, the chosen display name from the user profile, and the user's slug.
The text for the slug has too low contrast on the item in the list that is selected.
It is `#757575` on `#007cba` (contrast ratio 1:1) and hover it is `#007cba` on `#006ba1` (1.26:1).
`#007cba` and `#006ba1` are from the admin theme.

### Step-by-step reproduction instructions

Type `@` in the editor and select a user. Try to read the slug, which is the second name on the listed item.

### Screenshots, screen recording, code snippet

![A screenshot of the list of users, showing the text and background colors.](https://github.com/user-attachments/assets/4a8f2572-eeaa-451c-b4b8-e10f82fdb9a1)

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [x] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

**Comment by SainathPoojary** [comment]
Yes @carolinan, I agree this indeed is a issue, I would like to work on this.

**Comment by im3dabasia** [comment]
Thank you, @carolinan, for raising this issue. I agree that this is an accessibility concern, and based on WCAG AA guidelines, we should have at least a contrast ratio of 4.5:1 for small text.

I propose changing the color from `#757575` to `#fff`, as used in the Publish button on the top bar. This would ensure consistency and resolve the accessibility issue.

**Comment by carolinan** [comment]
To be honest I would label this as a low priority, unless testing show that there are other elements of other autocompletes that also has contrast issues.

----
By setting the text color to white, the display name and the slug would be too similar, and it may be confusing because they would then have the same importance in the interface.
A parenthesis (in addition to the color change) may also not be suitable for de-emphasizing the slug, since the context is not exactly a sentence?

I found it confusing that the slug showed at all, but maybe it is important in some environments with many users.

**Comment by SainathPoojary** [comment]
Hey @carolinan, 

I was wondering if adding @ before the slug and reducing the font size to 0.7rem might help make it less prominent.

**Here are a few visual comparisons for your reference:**

With font size: `0.7rem` and `@`:

<img width="280" alt="Image" src="https://github.com/user-attachments/assets/ce2956f1-a8c2-4605-9ebf-c907b0bdb45e" />


Without font size adjustment:

<img width="255" alt="Image" src="https://github.com/user-attachments/assets/22c441c2-267e-462a-8faa-5439f3808b8b" />


With font size `0.7rem` but without the `@`:

<img width="290" alt="Image" src="https://github.com/user-attachments/assets/d2929dd9-23bf-4653-b2e5-8aa6cf74ab4b" />

**Comment by afercia** [comment]
> maybe it is important in some environments with many users.

Yes I guess the slug helps differentiate users with similar usernames.

Due to my low vision I was really struggling to even _understand_ the screenshot attached to this issue because to me the slug of the selected item is barely visible. 

It is worth noting tha the color change of the selected item comes from the fact the button of the selected item becomes a `primary` button:

https://github.com/WordPress/gutenberg/blob/693e315bd00ac36828364bc0b0d8f1fe22b5963c/packages/components/src/button/style.scss#L54-L55

These colors use CSS variables that adapt to the admin color scheme set under the WP admin user profile settings. Instead, the user slug color [is hardcoded](https://github.com/WordPress/gutenberg/blob/693e315bd00ac36828364bc0b0d8f1fe22b5963c/packages/editor/src/components/autocompleters/style.scss#L27) in the component.

Screenshot with the default admin color scheme, 'Midnight' color scheme and 'Ectoplasm' color scheme

![Image](https://github.com/user-attachments/assets/f6437363-af2c-4989-9a91-0f6685527600)

While only the default admin color scheme is expected to be accessible, the background colos can't be predicted. As such, I think the slug color should use the same variable of the username color, maybe deemphasized with some Sass function?

**Comment by Rishit30G** [comment]
Thanks for sharing the issue, 

Since I was able to reproduce the issue, I found that using `color: currentColor;` instead of `color: var(--wp-admin-theme-color);` for the hover state resolves the problem. Also, setting the text color to something like `color: rgb(191, 189, 189); ` instead of `color: $gray-700;` makes the username much more visible.

Sharing the screencast for the same: 


https://github.com/user-attachments/assets/4c47b083-9e0b-480d-9b1a-66edabc6d976

**Comment by SainathPoojary** [comment]
Hi @Rishit30G,

I tested the suggested color using the [ColorContrastChecker](https://wordpress.github.io/gutenberg/?path=/story/blockeditor-contrastchecker--default&args=backgroundColor:!hex(007cba);textColor:!hex(bfbdbd)) component in Storybook, and unfortunately, it did not pass the contrast check. Based on my findings, #FFFFFF(which is same color as name) was the only shade of gray that met the accessibility requirements.

However, if we use the same color for both the name and slug, they become indistinguishable. I think we need to explore a way to visually de-emphasize the slug to differentiate it better.

Looking forward to hearing your thoughts!

**Comment by Rishit30G** [comment]
Hey @SainathPoojary, 

It makes sense, thanks for sharing your thoughts on the above screencast! 👍🏻 

For the hover part I think that we have `wp-admin-theme-color` on the hover ( non-selected ) so it makes sense to have the `currentColor` for the hover state ( selected ) for username text, that should be fine in my opinion 

However for the non-selected/selected state, the username can have a colour somewhere in the lines of `rgb(191, 189, 189);` that was my thinking, but if it fails the accessibility criteria, then we should go with the approach shared in the [above comment](https://github.com/WordPress/gutenberg/issues/69001#issuecomment-2630685671) ( deemphasizing using sass function )

**Comment by SainathPoojary** [comment]
Hi @afercia,

Thank you for your suggestion. When using SaaS classes to deemphasize the user slug by reducing opacity or blending colors, it seems to result in lower contrast.

Here’s an example of without deemphasized autocomplete appears:

https://github.com/user-attachments/assets/15be9040-2884-4902-a832-668d1518f0f7

Would it be worth considering any of the following improvements?

- Adding an @ before the slug
- Reducing the font size

Or does the current implementation already work well since the slug is in lowercase?

I’d appreciate your thoughts on this. 

Thank you.

**Comment by afercia** [comment]
Rather than attempting to try a 'de-emphasize' color I would simplify and just use white.
After all, the white would be used only for the 'highlighted' state. The default state would be unchanged.
To me, it's perfectly ok that the slug is not de-emphasized in the highlighted state.

**Comment by rotemg-elementor** [comment]
I agree with @afercia, I think changing to white would probably be the best.
